code/src/terrat_files/github/sql/select_dirspaces_owned_by_other_pull_requests.sql

```diff
@@ -50,103 +50,94 @@ all_necessary_dirspaces as (
         on gpr.repository = prapc.repository and gpr.pull_number = prapc.pull_number
     where gpr.repository = $repository and gpr.pull_number = $pull_number
 ),
-unmerged_pull_requests_with_applies as (
-    select distinct
+overlapping_pull_requests as (
+    select
         gpr.repository as repository,
         gpr.pull_number as pull_number
     from github_pull_requests as gpr
-    inner join github_work_manifests as gwm
-        on gwm.repository = gpr.repository and gwm.pull_number = gpr.pull_number
-    left join github_pull_request_latest_unlocks as gpru
-        on gpru.repository = gpr.repository and gpru.pull_number = gpr.pull_number
-    where gwm.repository = $repository
-          and gpr.state in ('open', 'closed') and gwm.run_type in ('apply', 'autoapply', 'unsafe-apply')
-          and (gpru.unlocked_at is null or gpru.unlocked_at < gwm.created_at)
-),
--- All those dirspaces that are in a pull request that has at least one apply
--- and not merged.
-dangling_dirspaces as (
-    select
-        ands.repository as repository,
-        ands.pull_number as pull_number,
-        ands.path as path,
-        ands.workspace as workspace
-    from unmerged_pull_requests_with_applies as uprwa
+    inner join github_change_dirspaces as gcds
+        on gcds.repository = gpr.repository
+           and gcds.base_sha = gpr.base_sha
+           and gcds.sha = gpr.sha
```

Comment:
Why do we need both of these?

Reply:
The same change can have multiple PRs that go to different destination branches.  This guarantees that the source and destination refs are the same as the pull requset's